action_skill_analysis_cp4d.ipynb

```diff
@@ -0,0 +1,995 @@
+    "import ibm_watson\n",
+    "\n",
+    "# Internal python libraries\n",
+    "from assistant_skill_analysis.utils import skills_util, lang_utils\n",
+    "from assistant_skill_analysis.highlighting import highlighter\n",
+    "from assistant_skill_analysis.data_analysis import summary_generator\n",
+    "from assistant_skill_analysis.data_analysis import divergence_analyzer\n",
+    "from assistant_skill_analysis.data_analysis import similarity_analyzer\n",
+    "from assistant_skill_analysis.term_analysis import chi2_analyzer\n",
+    "from assistant_skill_analysis.term_analysis import keyword_analyzer\n",
+    "from assistant_skill_analysis.term_analysis import entity_analyzer\n",
+    "from assistant_skill_analysis.confidence_analysis import confidence_analyzer\n",
+    "from assistant_skill_analysis.inferencing import inferencer\n",
+    "from assistant_skill_analysis.experimentation import data_manipulator"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Assistant Settings\n",
+    "Please set values for the variables in the cell below to configure this notebook. The notebook uses CloudPakForDataAuthenticator to authenticate the APIs.\n",
+    "\n",
+    "- **LANGUAGE_CODE:** language code correspond to your workspace data, supported languages: **en, fr, de, es, cs, it, pt, nl**\n",
+    "\n",
+    "- **ASSISTANT_ID:** id of the Watson Assistant service instance\n",
+    "\n",
+    "- **ACTION_SKILL_FILENAME:** Download action skill workspace json from UI, and set variable in the cell below to the downloaded path.\n",
+    "\n",
+    "- **username and password:** Replace them with your Cloud Pak for Data credentials\n",
+    "- **CP4D_URL:** This is the base url of your instance. It is in the format of `https://{cpd_cluster_host}{:port}/icp4d-api`\n",
+    "- **SERVICE_URL:** This is the service URL of your Watson Assistant. The URL follows this pattern: `https://{cpd_cluster_host}{:port}/assistant/{release}/instances/{instance_id}/api`. To find this URL, view the details for the service instance from the Cloud Pak for Data web client. For more information, see [Service Endpoint](https://cloud.ibm.com/apidocs/assistant-data-v1?code=python#service-endpoint)\n",
+    "\n",
+    "Reference: https://cloud.ibm.com/apidocs/assistant-v2#authentication\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# language\n",
+    "\n",
+    "LANGUAGE_CODE = \"en\"\n",
+    "\n",
+    "lang_util = lang_utils.LanguageUtility(LANGUAGE_CODE)\n",
+    "\n",
+    "_, _, ASSISTANT_ID = skills_util.input_credentials(input_apikey=False,input_workspace_id=False,input_assistant_id=True)\n",
+    "\n",
+    "username = 'apikey'\n",
+    "password = '###'\n",
+    "SERVICE_URL = 'service_url'\n",
+    "CP4D_URL = 'cp4d_url'\n",
+    "BEARER_TOKEN = 'bearer_token' # this is optional\n",
```

Comment:
let's move this right above the line of `skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN, url=SERVICE_URL)`